File: lab2-functionalprimer/python3/func_as_arg.py
from typing import List


def listFunc(a:int, b:int) -> List[int]:
    return [i for i in range(a, b+1)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]

def applicatorFunc(inpFunc,a:int,b:int,s:str):
    if s=='s':
        return sum(inpFunc(a,b))
    else:
        return sum(inpFunc(a,b))/(b-a+1)

print(applicatorFunc(listFunc,1,10, 'a'))

File: lab2-functionalprimer/javascript/func_as_arg.js
//Create list of ints from 1 to 5, Haskell equivalent [1..5]
function arrFunc(a, b){
    let arr = [];
    for (let i = a; i<=b; i++) {
        arr.push(i);
    }
    return arr;    
}

function applicatorFunc(inpFunc ,a, b, s){
    const arr = inpFunc(a, b);   
    let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);    
    if(s=='s'){
        return sum;
    }
    else{        
        return sum/(b-a+1);
    }
}

let x = applicatorFunc(arrFunc,1,10, 'a');
console.log(x);